src/routes/class-base-route.php

```diff
@@ -123,12 +123,21 @@ protected function get_route_name() : string {
    */
   protected function get_callback_arguments() : array {
     return [
-      'methods'  => static::READABLE,
+      'methods'  => $this->get_methods(),
       'callback' => [ $this, 'route_callback' ],
       'permission_callback' => [ $this, 'permission_callback' ],
     ];
   }
 
+  /**
+   * Returns allowed methods for this route.
+   *
+   * @return string|array
+   */
+  protected function get_methods() {
```

Comment:
Why method? I'd leave this as a constant, that's the way we always do it. If you need multiple methods, you can set them as a constant either as an array, or a comma-separated string.